packages/cozy-interapp/README.md

```diff
@@ -4,21 +4,114 @@ Library to create intents : actions that need to be fullfilled by another app th
 
 [More info on intents here](https://docs.cozy.io/en/cozy-stack/intents/#intents)
 
-## Usage
+### `cozy.client.intents.create()`
+
+`cozy.client.intents.create(action, doctype [, data, permissions])` create an intent. It returns a modified Promise for the intent document, having a custom `start(element)` method. This method interacts with the DOM to append an iframe to the given HTML element. This iframe will provide an access to an app, which will serve a service page able to manage the intent action for the intent doctype. The `start(element)` method returns a promise for the result document provided by intent service.
+
+> __On Intent ready callback:__ This `start` method also takes a second optional argument which is a callback function (`start(element, onReadyCallback)`). When provided, this function will be run when the intent iframe will be completely loaded (using the `onload` iframe listener). This callback could be useful to run a client code only when the intent iframe is ready and loaded.
+
+An intent has to be created everytime an app need to perform an action over a doctype for wich it does not have permission. For example, the Cozy Drive app should create an intent to `pick` a `io.cozy.contacts` document. The cozy-stack will determines which app can offer a service to resolve the intent. It's this service's URL that will be passed to the iframe `src` property.
+
+Once the intent process is terminated by service, the iframe is removed from DOM.
+
+#### Example
+```js
+cozy.client.intents.create('EDIT', 'io.cozy.photos', {action: 'crop', width: 100, height: 100})
+  .start(document.getElementById('intent-service-wrapper'))
+```
+
+See cozy-stack [documentation](https://docs.cozy.io/en/cozy-stack/intents/) for more details.
+
+You can also use `.then` to run some code after the intents is terminated like following:
 
 ```js
-import CozyClient from 'cozy-client'
-import Intents from 'cozy-interapp'
-
-const client = new CozyClient({ uri, token })
-const intents = new Intents({ client })
-intents.create('io.cozy.apps', 'CREATE')
-  .start()
-  .then(intent => {
-    // do something with the result of the intent
-    // intent = {
-    //    id: '1337',
-    //    attributes: {...}
-    // }
+cozy.client.intents.create('EDIT', 'io.cozy.photos', {action: 'crop', width: 100, height: 100})

```

Comment:
The `cozy.client.intents` is from another age where the intents were integrated in the client. The right way to instantiate the `Intents` is the one that has been removed here:

```
const client = new CozyClient({ uri, token })
const intents = new Intents({ client })
intents.create('io.cozy.apps', 'CREATE')
```

I think it is sufficient to replace

```patch
- cozy.client.intents.create('EDIT', 'io.cozy.photos', {action: 'crop', width: 100, height: 100})
+ const client = new CozyClient({ uri, token })
+ const intents = new Intents({ client })
+ intents.create('io.cozy.apps', 'CREATE')
```

The EDIT intent for io.cozy.photos does not exist so let's not put it in the doc :)

Reply:
Ok :) Cela semble être plutôt `intents.create( 'CREATE', 'io.cozy.apps')` (cf https://github.com/cozy/cozy-libs/blob/doc/intent/packages/cozy-interapp/src/intents.js#L12 et https://github.com/cozy/cozy-ui/blob/feat/Intents/react/IntentIframe/IntentIframe.jsx#L39)